src/lib/run-dev-local.js

```diff
@@ -0,0 +1,122 @@
+/*
+Copyright 2020 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+const path = require('path')
+const fs = require('fs-extra')
+const cloneDeep = require('lodash.clonedeep')
+const utils = require('./app-helper')
+const dedent = require('dedent-js')
+const rtLib = require('@adobe/aio-lib-runtime')
+const rtLibUtils = rtLib.utils
+const aioLogger = require('@adobe/aio-lib-core-logging')('@adobe/aio-cli-plugin-app:run-dev-local', { provider: 'debug' })
+
+const {
+  OW_CONFIG_RUNTIMES_FILE,
+  OW_JAR_URL, OW_JAR_PATH,
+  OW_LOCAL_APIHOST,
+  OW_LOCAL_NAMESPACE,
+  OW_LOCAL_AUTH,
+  OW_LOCAL_LOG_FILE
+} = require('../lib/owlocal')
+
+const OW_WAIT_INIT_TIME = 2000
+const OW_WAIT_PERIOD_TIME = 500
+const OW_TIMEOUT = 60000
+
+/**
+ * @typedef {object} RunDevLocalObject
+ * @property {string} config the modified dev config
+ * @property {Function} cleanup callback function to cleanup available resources
+ */
+
+/**
+ * Checks the system for pre-requisites to run local Openwhisk, then runs it.
+ *
+ * @param {object} config the app config
+ * @param {Function} [log] function to log application logs
+ * @param {boolean} [verbose=false] set to true to have verbose logging (openwhisk)
+ * @returns {RunDevLocalObject} the RunDevLocalObject
+ */
+async function runDevLocal (config, log = () => undefined, verbose = false) {
```

Comment:
small comment: should we rename run-dev-local to smthg like run-local-runtime ? To emphasize that this is only handling the `backend`/`openwhisk` side of things (and not local UI serve)

Reply:
good idea, I'll rename.

Reply:
fixed in 51264c2